feat(ingress): Add native Traefik IngressRoute support

salehi opened the pull request:
# PR Title
feat(ingress): Add native Traefik IngressRoute support

## Description

This PR adds native Traefik `IngressRoute` CRD support as an alternative to standard Kubernetes `Ingress` resources.

### Problem

When using Traefik as the ingress controller, the current `Ingress` resource with regex paths does not work correctly. The chart generates:

```yaml
spec:
  ingressClassName: traefik
  rules:
  - host: sentry.example.com
    http:
      paths:
      - path: /api/[1-9][0-9]*/(.*)
        pathType: ImplementationSpecific
        backend:
          service:
            name: sentry-relay
            port:
              number: 3000
```

With nginx annotations:
```yaml
annotations:
  nginx.ingress.kubernetes.io/use-regex: "true"
```

**The problem:** Traefik ignores nginx-style annotations, so the regex paths are not matched correctly. All traffic falls through to the `/` catch-all route and is sent to `sentry-web` instead of `sentry-relay`.

This causes SDK event submission to fail with:
- 403 CSRF errors (hitting web UI login instead of relay)
- 302 redirects to login page

### Solution

Add native Traefik `IngressRoute` CRD support that properly handles regex path matching:

```yaml
apiVersion: traefik.io/v1alpha1
kind: IngressRoute
spec:
  routes:
    - match: Host(`sentry.example.com`) && PathRegexp(`/api/[0-9]+/`)
      kind: Rule
      priority: 100
      services:
        - name: sentry-relay
          port: 3000
```

### Changes

1. **`values.yaml`**: Added `ingressRoute` configuration section
2. **`templates/ingressroute.yaml`**: New template for Traefik IngressRoute CRDs
3. **`templates/_helpers.tpl`**: Added `relay.port` helper (if not present)
4. **`README.md`**: Documentation for Traefik IngressRoute usage

### Usage

```yaml
ingress:
  enabled: false  # Disable standard Ingress

ingressRoute:
  enabled: true
  hostname: sentry.example.com
  entryPoints:
    - web
    - websecure
  # tls:
  #   secretName: sentry-tls  # Optional
```

### Testing

Tested with:
- Traefik v2.x / v3.x on k3s
- Sentry 25.9.0
- Self-hosted Sentry Helm chart 27.8.0

Verified:
- `/api/2/store/` returns 401 (relay responding) instead of 302 (web redirecting)
- SDK events successfully ingested
- Web UI accessible at `/`
- Relay-to-relay communication working at `/api/0/relays/`

### Related Issues

<!-- Link any related issues here -->
- Fixes routing issues when using Traefik as ingress controller

### Checklist

- [x] Code follows project style guidelines
- [x] Documentation updated
- [x] Tested with Traefik ingress controller
- [x] Backwards compatible (new feature, opt-in)

salehi committed:
feat(ingress): Add native Traefik IngressRoute support

patsevanton commented:
Why traefik? Maybe add gateway API?

cynecx commented:
We've had issues because our cluster uses traefik v3 and the rule syntax has changed. The workaround for that was to fallback to the v2 rulesyntax:

```
ingress:
  enabled: true
  regexPathStyle: traefik
  ingressClassName: traefik
  annotations:
    # Make the ingress work with Traefik v3
    traefik.ingress.kubernetes.io/router.rulesyntax: v2
```

This seems to work quite well for us. The obvious caveat is that we don't know how long they are gonna keep the fallback working and not fully deprecate it.

salehi commented:
> Why traefik? Maybe add gateway API?

I just deployed to k3s, the default it traefik.

salehi commented:
I'm using these changes for compatibility with k3s and just wanted to share the work with the main stream.

Mokto commented on charts/sentry/values.yaml:
```diff
@@ -2401,6 +2401,36 @@ ingress:
   # - secretName:
   #   hosts:
 
+ingressRoute:
```
I think you should call it "traefikIngressRoute" ?

salehi replied:
Yes, you are right.
Done.

salehi committed:
fix(ingress): rename ingressRoute to traefikIngressRoute

Disambiguates the Traefik-specific IngressRoute CRD from generic ingress
configuration to avoid confusion with standard Kubernetes Ingress.

salehi commented:
force pushed for rebase from develop.
